airbyte-scheduler/models/src/main/java/io/airbyte/scheduler/models/JobStatus.java

```diff
@@ -19,4 +20,12 @@ public enum JobStatus {
   public static final Set<JobStatus> TERMINAL_STATUSES = Sets.newHashSet(FAILED, SUCCEEDED, CANCELLED);
   public static final Set<JobStatus> NON_TERMINAL_STATUSES = Sets.difference(Set.of(values()), TERMINAL_STATUSES);
 
+  public static final Map<JobStatus, Set<JobStatus>> VALID_STATUS_CHANGES = Map.of(
+      PENDING, Set.of(RUNNING, FAILED, CANCELLED),
+      RUNNING, Set.of(INCOMPLETE, SUCCEEDED, FAILED, CANCELLED),
+      INCOMPLETE, Set.of(PENDING, RUNNING, FAILED, CANCELLED),
+      SUCCEEDED, Set.of(),
+      FAILED, Set.of(SUCCEEDED),
```

Comment:
Can we go from failed/cancelled to succeeded? I though there were terminal. I saw the graph and I don't think it should happen

Reply:
same thought. If `cancelled`, `failed` and `succeeded` are terminal, I'd think that set of valid transitions should be null.

Reply:
Took another look at this, and I agree with both of you. These were status transitions leftover from the old scheduler days, and it doesn't look like they are able to occur now in the temporal scheduler. I will remove these